crd-generator/test/src/test/java/io/fabric8/crd/generator/types/TypeMappingsTest.java

```diff
@@ -0,0 +1,90 @@
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+class TypeMappingsTest {
+
+  private CustomResourceDefinition crd;
+
+  @BeforeEach
+  void setUp() {
+    crd = Serialization.unmarshal(
+        TypeMappingsTest.class.getResourceAsStream("/META-INF/fabric8/typemappings.type-mappings.fabric8.io-v1.yml"),
+        CustomResourceDefinition.class);
+  }
+
+  @ParameterizedTest(name = "{0} type maps to {1}")
+  @MethodSource("targetTypeCases")
+  void targetType(String propertyName, String expectedType) {
+    assumeTrue(System.getProperty("java.version").startsWith("1.8")); // TODO: remove after #5877
+    assertThat(crd.getSpec().getVersions().iterator().next().getSchema().getOpenAPIV3Schema().getProperties())
+        .extracting(schema -> schema.get("spec").getProperties())
+        .returns(expectedType, specProps -> specProps.get(propertyName).getType());
+  }
+
+  private static Stream<Arguments> targetTypeCases() {
+    return Stream.of(
+        Arguments.of("date", "string"),
+        Arguments.of("localDate", "object"), // to review
+        Arguments.of("localDateTime", "object"), // to review
+        Arguments.of("zonedDateTime", "object"), // to review
+        Arguments.of("offsetDateTime", "object"), // to review
+        Arguments.of("offsetTime", "object"), // to review
+        Arguments.of("yearMonth", "object"), // to review
+        Arguments.of("monthDay", "object"), // to review
+        Arguments.of("instant", "object"), // to review
+        Arguments.of("duration", "object"), // to review
+        Arguments.of("period", "object"), // to review
+        Arguments.of("timestamp", "object"), // to review
+        // Arguments.of("aShort", "integer"), // TODO: Not even present in the CRD
+        Arguments.of("aShortObj", "object"), // to review
+        Arguments.of("aInt", "integer"),
+        Arguments.of("aIntegerObj", "integer"),
+        Arguments.of("aLong", "integer"),
+        Arguments.of("aLongObj", "integer"),
+        Arguments.of("aDouble", "number"),
+        Arguments.of("aDoubleObj", "number"),
+        Arguments.of("aFloat", "number"),
+        Arguments.of("aFloatObj", "number"),
+        Arguments.of("aNumber", "object"), // to review
+        Arguments.of("aBigInteger", "object"), // to review
+        Arguments.of("aBigDecimal", "object"), // to review
+        Arguments.of("aBoolean", "boolean"),
+        Arguments.of("aBooleanObj", "boolean"),
+        // Arguments.of("aChar", "string"), // TODO: Not even present in the CRD
+        Arguments.of("aCharacterObj", "object"), // to review
+        Arguments.of("aCharArray", "array"),
+        Arguments.of("aCharSequence", "object"), // to review
+        Arguments.of("aString", "string"),
+        Arguments.of("aStringArray", "array"),
+        // Arguments.of("aByte", "?"), // TODO: Not even present in the CRD
+        Arguments.of("aByteObj", "object"), // to review
+        Arguments.of("aByteArray", "array"));
```

Comment:
```suggestion
        Arguments.of("aByteArray", "array")); // to review
```

byte[] should default to a base64 encoded string.

Reply:
@manusa that's why the suggestion is to add the to review comment

Reply:
yes, yes, sorry. I realized when adding the uuid field